src/ansys/units/unit.py

```diff
@@ -331,6 +331,31 @@ def _new_units(self, __value, op):
 
         return Unit(self._condense(new_units))
 
+    def _temp_precheck(self, other_unit, op: str = None):
+        """
+        Validate units for temperature differences.
+
+        Parameters
+        ----------
+        other_unit : Unit
+            Unit for comparison against current unit.
+        op : str, optional
+            Operation conducted on the units. "-"
+
+        Returns
+        -------
+        Unit | None
+            unit object for a quantity of temperature difference or temperature.
+        """
+
+        temp = Unit("K")
+        delta_temp = Unit("delta_K")
+        if (self and other_unit) in (temp, delta_temp):
+            if self != other_unit:
+                return Unit(self.name[-1])
```

Comment:
I guess it's everything after the delta you're after, so the logic is not solid here. Consider that that the suffix can be of any length.

Reply:
Updated to handle any length suffix.